test-data/unit/check-dataclasses.test

```diff
@@ -1536,3 +1556,31 @@ A(a=1, b=2)
 A(1)
 A(a="foo")  # E: Argument "a" to "A" has incompatible type "str"; expected "int"
 [builtins fixtures/dataclasses.pyi]
+
+[case testDataclassesFieldWithDefaultOverride]
+# flags: --python-version 3.7
+from dataclasses import dataclass
+
+@dataclass
+class A:
+    a: int = 0
+
+@dataclass
+class Correct1(A):
+    a: int = 0
+@dataclass
+class Correct2(A):
+    a: int = 1
+    b: int = 2
+
+@dataclass
+class Wrong1(A):  # E: Attributes without a default cannot follow attributes with one
```

Comment:
Interestingly the runtime accepts this one. It's pretty weird, though, and I'm OK with mypy giving an error for it.